site/content/extend/plugins/developer-workflow.md

```diff
@@ -0,0 +1,66 @@
+---
+title: "Developer Workflow"
+date: 2020-07-11T23:00:00-04:00
+weight: 3
+subsection: Plugins
+---
+
+### Common `make` commands
+
+- `make dist` - Compile the plugin into a g-zipped file, ready to upload to a Mattermost server. The file is saved in the plugin repo's `dist` folder.
+- `make deploy` - Compiles the plugin using the `make dist` command, then automatically deploys the plugin to the Mattermost server
+- `make watch` - Uses webpack's watch feature to re-compile and deploy the webapp portion of your plugin on any change to the `webapp/src` folder
+- `make test` - Runs the plugin's server tests and webapp tests
+- `make check-style` - Runs linting checks on the plugin's server and webapp folders
+- `make enable` - Enables the plugin on the Mattermost server
+- `make disable` - Disables the plugin on the Mattermost server
+- `make reset` - Disables and re-enables the plugin on the Mattermost server
+- `make debug-plugin` - Starts a `delve` process and attaches it to your running plugin
+
+You can run the development build of the plugin by setting the environment variable `MM_DEBUG=1`, or prefixing the variable at the beginning of the `make` command. For example, `MM_DEBUG=1 make deploy` will deploy the development build of the plugin to your server, allowing you to have a more fluid debugging experience. To use the production build of the plugin instead, unset the `MM_DEBUG` environment variable before running the `make` commands.
+
+### Exposing the Mattermost server using `ngrok`
```

Comment:
Maybe we should mention https://localhost.run as an alternative for ngrok? I prefer this one because you don't need to install anything. You just have to run a command, e.g.:
`ssh -R 80:localhost:8065 ssh.localhost.run`
I was using it when developing S4B plugin and it worked very well :+1: .

Reply:
How does this work if there is an external service involved like Jira Cloud or GitHub? Can that service access your local server through `localhost.run`?

Reply:
Sure. It works like ngrok. For example, the result of this command:
`ssh -R 80:localhost:8065 ssh.localhost.run`
Will be this:
`Connect to http://grzegorz-d8490c28.localhost.run`
By using the address `grzegorz-d8490c28.localhost.run`, external services have access to your local server running on port 8065.

Reply:
@kosgrz This is really cool! It even supports an `https` endpoint. I will still recommend ngrok due to the debugging tools it gives you to analyze the requests/responses, but I will mention this tool as well. Thanks for mentioning this!